collectors/debugfs.plugin/debugfs_zswap.c

```diff
@@ -0,0 +1,117 @@
+
+    const char *chart_id;
+    const char *dimension;
+    const char *units;
+    const char *title;
+
+    int enabled;
+    int chart_created;
+
+    int prio;
+} zswap_metrics[] = {
+    // https://elixir.bootlin.com/linux/latest/source/mm/zswap.c
+    {.filename = "/sys/kernel/debug/zswap/same_filled_pages", .chart_id = "same_filled_page",
+     .dimension = "pages", .units = "pages", .title = "Total same-value filled pages currently stored",
+     .enabled = CONFIG_BOOLEAN_YES, .chart_created = CONFIG_BOOLEAN_NO,
+     .prio = NETDATA_CHART_PRIO_SYSTEM_ZSWAP_SAME_FILL_PAGE},
+    {.filename = "/sys/kernel/debug/zswap/stored_pages", .chart_id = "stored_pages",
+     .dimension = "page", .units = "pages", .title = "Compressed pages stored in zswap.",
+     .enabled = CONFIG_BOOLEAN_YES, .chart_created = CONFIG_BOOLEAN_NO,
+     .prio = NETDATA_CHART_PRIO_SYSTEM_ZSWAP_STORED_PAGE},
+    {.filename = "/sys/kernel/debug/zswap/pool_total_size", .chart_id = "pool_total_size",
+     .dimension = "pool", .units = "bytes", .title = "Total bytes used by the compressed storage",
+     .enabled = CONFIG_BOOLEAN_YES, .chart_created = CONFIG_BOOLEAN_NO,
+     .prio = NETDATA_CHART_PRIO_SYSTEM_ZSWAP_POOL_TOT_SIZE},
+    {.filename = "/sys/kernel/debug/zswap/duplicate_entry", .chart_id = "duplicate_entry",
+     .dimension = "duplicate", .units = "page", .title = "Duplicate store was found.",
+     .enabled = CONFIG_BOOLEAN_YES, .chart_created = CONFIG_BOOLEAN_NO,
+     .prio = NETDATA_CHART_PRIO_SYSTEM_ZSWAP_DUPP_ENTRY},
+    {.filename = "/sys/kernel/debug/zswap/written_back_pages", .chart_id = "written_back_pages",
+     .dimension = "pages", .units = "pages", .title = "Pages written back when pool limit was reached.",
+     .enabled = CONFIG_BOOLEAN_YES, .chart_created = CONFIG_BOOLEAN_NO,
+     .prio = NETDATA_CHART_PRIO_SYSTEM_ZSWAP_STORED_PAGE},
+    {.filename = "/sys/kernel/debug/zswap/reject_compress_poor", .chart_id = "reject_compress_poor",
+     .dimension = "pages", .units = "pages", .title = "Compressed page was too big for the allocator to store.",
+     .enabled = CONFIG_BOOLEAN_YES, .chart_created = CONFIG_BOOLEAN_NO,
+     .prio = NETDATA_CHART_PRIO_SYSTEM_ZSWAP_REJECT_COM_POOR},
+    {.filename = "/sys/kernel/debug/zswap/reject_kmemcache_fail", .chart_id = "reject_kmemcache_fail",
+     .dimension = "pages", .units = "pages", .title = "Number of entry metadata that could not be allocated",
+     .enabled = CONFIG_BOOLEAN_YES, .chart_created = CONFIG_BOOLEAN_NO,
+     .prio = NETDATA_CHART_PRIO_SYSTEM_ZSWAP_KMEM_FAIL},
+    {.filename = "/sys/kernel/debug/zswap/reject_alloc_fail", .chart_id = "reject_alloc_fail",
+     .dimension = "allocator", .units = "calls", .title = "Allocator could not get memory",
+     .enabled = CONFIG_BOOLEAN_YES, .chart_created = CONFIG_BOOLEAN_NO,
+     .prio = NETDATA_CHART_PRIO_SYSTEM_ZSWAP_RALLOC_FAIL},
+    {.filename = "/sys/kernel/debug/zswap/reject_reclaim_fail", .chart_id = "reject_reclaim_fail",
```

Comment:
I think these 4 should be dimensions. All of them mean the number of rejected pages => aggregation (sum of all dimensions) makes sense.

Reply:
And if you don't mind please apply formatting at least to this array. The following is much easier to read when reviewing:

```c
    {.filename = "/sys/kernel/debug/zswap/same_filled_pages",
     .chart_id = "same_filled_page",
     .dimension = "pages",
     .units = "pages",
     .title = "Total same-value filled pages currently stored",
     .enabled = CONFIG_BOOLEAN_YES,
     .chart_created = CONFIG_BOOLEAN_NO,
     .prio = NETDATA_CHART_PRIO_SYSTEM_ZSWAP_SAME_FILL_PAGE},
```

Reply:
I accepted all suggestions, thank you!